File: mobile/android/app/src/main/java/ru/edu/asu/minobrlabs/sensors/common/AbstractSensorsWorker.java
package ru.edu.asu.minobrlabs.sensors.common;

import android.util.Log;

import ru.edu.asu.minobrlabs.App;

public abstract class AbstractSensorsWorker<T extends ISensorsManager> extends AbstractWorker {
    private final String tag;

    private final T sensors;

    public AbstractSensorsWorker(final T sensors, final String tag) {
        super();

        this.sensors = sensors;
        this.tag = tag;
    }

    @Override
    public void start() {
        sensors.start();

        super.start();
    }

    @Override
    public void run() {
        try {
            while (running) {
                sensors.update();

                sleep(App.state.storage.sleepTime);
            }
        } catch (InterruptedException e) {
            Log.e(tag, "Sensors worker problems", e);
        }
    }

    @Override
    public void kill() {
        super.kill();
    }
}

File: desktop/windows/MinobrLabsProject/MinobrLabsProject/sensors/common/AbstractSensorsWorker.cs
using System;
using System.Threading;

namespace MinobrLabsProject.sensors.common
{
    public abstract class AbstractSensorsWorker<T> : AbstractWorker where T : ISensorsManager
    {
        private readonly T sensors;

        public AbstractSensorsWorker(T sensors) : base()
        {
            this.sensors = sensors;
        }
        
        public override void run()
        {
            try
            {
                while (running)
                {
                    sensors.update();
                    sleep();
                }
            }
            catch (Exception e)
            {
                Console.WriteLine("Sensors worker problems", e);
            }
        }

        protected void sleep()
        {
            Thread.Sleep(200); //TODO Брать время из конфига
        }
    }
}
